internal/commands/main.go

```diff
@@ -162,22 +169,14 @@ func (c Commands) FindArticle(substr string) (item rss.Item, err error) {
 		return rss.Item{}, fmt.Errorf("commands.FindArticle: %w", err)
 	}
 
-	regex, err := regexp.Compile(strings.ToLower(substr))
-	if err != nil {
-		return rss.Item{}, fmt.Errorf("commands.FindArticle: regexp: %w", err)
-	}
-
 	for _, r := range rsss {
-		for _, it := range r.Channel.Items {
-			// very basic string matching on title to read an article
-			if regex.MatchString(strings.ToLower(it.Title)) {
-				item = it
-				break
-			}
+		item, err := r.FindArticle(substr)
+		if item != nil || err != nil {
+			return *item, err
```

Comment:
General pattern in go is to check for an error and explitly return, `<zero_value>, err` if therre is an error.

This would be better more explicitly as:
```go
if err != nil {
  return rss.Item{}, err 
}

if item == nil {
  return rss.Item{}, errors.New(fmt.Sprintf("Article %s not found", substr))
}
...
return *item, nil
```

Reply:
@guyfedwards noted. Cheers!